src/main/java/com/google/vinet/data/MatchingRunner.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.vinet.data;
+
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.util.*;
+
+public class MatchingRunner {
+  private DatastoreService datastore;
+  private Set<IsolateTimeSlot> isolateTimeSlots;
+  private Set<VolunteerTimeSlot> volunteerTimeSlots;
+  private LocalDate date;
+
+  /**
+   * Construct a MatchingRunner with none of its data set.
+   */
+  public MatchingRunner() {
+
+  }
+
+  /**
+   * Construct a matching runner with a dependency on the provided DataStore implementation.
+   * @param datastore The DataStore implementation that this MatchingRunnner will depend on.
+   */
+  protected MatchingRunner(DatastoreService datastore) {
+    assert(datastore != null);
+    this.datastore = datastore;
+  }
+
+  /**
+   * Run the matching algorithm and store the results in DataStore.
+   * Any necessary data that is not already set will be pulled from DataStore.
+   */
+  public void run() {
+    if (this.datastore == null) this.datastore = DatastoreServiceFactory.getDatastoreService();
+    if (this.date == null) this.date = LocalDate.now().plusDays(1);
+    if (this.isolateTimeSlots == null) this.isolateTimeSlots = this.fetchIsolateTimeSlots();
+    if (this.volunteerTimeSlots == null) this.volunteerTimeSlots = this.fetchVolunteerTimeSlots();
```

Comment:
Why the use of `this` keywords if there aren't local variables of the same name?

Reply:
It was intended to improve clarity when reading the code but I've removed it now https://github.com/googleinterns/step226-2020/pull/10/commits/447670b6e8ba617d0e416ff110726ede26c634c9